examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2077 @@
+                config.NUM_PROCESSES, 1, device=self.device
+            )
+
+            rgb_frames = [
+                [] for _ in range(self.config.NUM_PROCESSES)
+            ]  # type: List[List[np.ndarray]]
+
+            if len(config.VIDEO_OPTION) > 0:
+                os.makedirs(config.VIDEO_DIR, exist_ok=True)
+
+            self.actor_critic.eval()
+
+            for i in range(config.TASK_CONFIG.ENVIRONMENT.MAX_EPISODE_STEPS):
+                current_episodes = envs.current_episodes()
+
+                with torch.no_grad():
+                    (
+                        _,
+                        actions,
+                        _,
+                        test_recurrent_hidden_states,
+                    ) = self.actor_critic.act(
+                        batch,
+                        test_recurrent_hidden_states,
+                        prev_actions,
+                        not_done_masks,
+                        deterministic=False,
+                    )
+
+                    prev_actions.copy_(actions)
+
+                outputs = envs.step([a[0].item() for a in actions])
+
+                observations, rewards, dones, infos = [
+                    list(x) for x in zip(*outputs)
+                ]
+                batch = batch_obs(observations, device=self.device)
+
+                not_done_masks = torch.tensor(
+                    [[0.0] if done else [1.0] for done in dones],
+                    dtype=torch.float,
+                    device=self.device,
+                )
+
+                rewards = torch.tensor(
+                    rewards, dtype=torch.float, device=self.device
+                ).unsqueeze(1)
+
+                current_episode_reward += rewards
+
+                # episode ended
+                if not_done_masks[0].item() == 0:
+                    generate_video(
+                        video_option=self.config.VIDEO_OPTION,
+                        video_dir=self.config.VIDEO_DIR,
+                        images=rgb_frames[0],
+                        episode_id=current_episodes[0].episode_id,
+                        checkpoint_idx=0,
+                        metrics=self._extract_scalars_from_info(infos[0]),
+                        tb_writer=None,
+                    )
+
+                    print("Evaluation Finished.")
+                    print("Success: {}".format(infos[0]["episode_success"]))
+                    print(
+                        "Reward: {}".format(current_episode_reward[0].item())
+                    )
+                    print(
+                        "Distance To Goal: {}".format(
+                            infos[0]["object_to_goal_distance"]
+                        )
+                    )
+
+                    return
+
+                # episode continues
+                elif len(self.config.VIDEO_OPTION) > 0:
+                    frame = observations_to_image(observations[0], infos[0])
+                    rgb_frames[0].append(frame)
+
+
+# %%
```

Comment:
Why is this a separate cell, you could merge it with the next cell if you wanted to.

Reply:
I wanted to keep the tensoboard output different from the logs which get generated in the next cell